modules/video/test/ocl/test_optflowpyrlk.cpp

```diff
@@ -121,13 +124,24 @@ OCL_TEST_P(PyrLKOpticalFlow, Mat)
             float errdiff = 0.0f;
 
             if (!eq || errdiff > 1e-1)
+            {
                 ++mistmatch;
+                continue;
+            }
+
+            eq = std::abs(cpuErr[i] - err[i]) < 0.01;
+            if(!eq &&
+               winSize.width >= 16 && winSize.height >=16 &&
+               winSize.width <=24 && winSize.height <=24)
```

Comment:
Hi alalek, 
I am totally get your point. This test not only works for OpenCL, but also works for the CUDA and any other optimizied version code. Therefore, this modification may badly influence the relevant code using this test.
In modules/video/src/opencl/pyrlk.cl file, however, the kernel can only supports the winSize within 16*16 to 24*24, although its input accepts the winSize outside this range. In this kernel, the input winSize bigger than 24*24 will be refered as 24*24, the input winSize smaller than 16*16 will be refered as 16*16. In the test case 2 and 3, with winSize 25*25, although the kernel can output the correct result of nextPts, the output of err is wrong even with the patch I pulled. This is the reason why I modify this test adding the winSize limitation in it.
So, are there any way to resolve this problem? How about that I add a new test specifically for OpenCL optical flow? Or would you mind that give me any other ideas?

Best,